### PR TITLE
Allow --list-rules with custom module set

### PR DESCRIPTION
**Pull request by bocekm**
Before, `--list-rules` option did not work with custom module set (`--contents` option). It was ignoring the passed custom module set path and instead it was using the module sets available in the `/usr/share/preupgrade` folder.

Resolves rhbz#[1362708](https://bugzilla.redhat.com/show_bug.cgi?id=1362708)
Resolves https://github.com/upgrades-migrations/preupgrade-assistant/issues/325
Resolves https://github.com/upgrades-migrations/preupgrade-assistant/issues/334

### CODE REVIEW

File: preupg/application.py [hunk 1]
```diff
@@ -459,48 +443,40 @@ def prepare_scan_system(self):
...
 
     def generate_report(self):
         """Function generates report"""
-        scenario = self.get_scenario()
-        scenario_path = os.path.join(self.conf.source_dir, scenario)
-        self.assessment_dir = os.path.join(self.conf.assessment_results_dir,
-                                           self.rename_custom_module_set(scenario))
-        dir_util.copy_tree(scenario_path, self.assessment_dir)
-        # Try copy directory with contents to /root/preupgrade
+        self.module_set_copy_path = os.path.join(
```
Comment (pirat89) [inline preupg/application.py]: it would be better set this inside the `determine_module_set_location` method. Or maybe right after call of the method.

File: preupg/application.py [hunk 2]
```diff
@@ -459,48 +443,40 @@ def prepare_scan_system(self):
...
-                    module_set_path = os.path.normpath(
-                        os.path.join(self.execution_dir, module_set_path))
-            ModuleSetUtils.get_module_set_os_versions(module_set_path)
+            ModuleSetUtils.get_module_set_os_versions(self.module_set_path)
         except EnvironmentError as err:
             log_message(str(err), level=logging.ERROR)
             return False
         return True
 
     def generate_report(self):
```
Comment (pirat89) [inline preupg/application.py]: it should be renamed in future as it does something completely different than it says.

File: preupg/application.py [hunk 3]
```diff
@@ -459,48 +443,40 @@ def prepare_scan_system(self):
...
-        self.assessment_dir = os.path.join(self.conf.assessment_results_dir,
-                                           self.rename_custom_module_set(scenario))
-        dir_util.copy_tree(scenario_path, self.assessment_dir)
-        # Try copy directory with contents to /root/preupgrade
+        self.module_set_copy_path = os.path.join(
+            settings.assessment_results_dir,
+            self.rename_custom_module_set(self.module_set_dirname))
+        dir_util.copy_tree(self.module_set_path,
+                           self.module_set_copy_path)
+        self.all_xccdf_xml_copy_path = os.path.join(
```
Comment (pirat89) [inline preupg/application.py]: Same as above.

File: preupg/application.py [hunk 4]
```diff
@@ -732,19 +737,17 @@ def run(self):
                     ' '.join(SystemIdentification.get_convertors())))
                 return ReturnValues.MISSING_TEXT_CONVERTOR
 
-        if not self.conf.scan and not self.conf.contents and \
-                not self.conf.list_rules:
-            self.conf.scan = self.get_default_module_set_dirname()
-            if not self.conf.scan:
-                return ReturnValues.SCENARIO
+        if self.conf.scan and self.conf.contents:
```
Comment (pirat89) [inline preupg/application.py]: This should be handled inside `preupg/cli.py`.

File: preupg/application.py [hunk 5]
```diff
@@ -779,56 +782,33 @@ def run(self):
...
+            return ReturnValues.MISSING_OPENSCAP
+        if not os.access(settings.openscap_binary, os.X_OK):
+            log_message("Oscap with SCE %s is not executable"
+                        % settings.openscap_binary)
+            return ReturnValues.MISSING_OPENSCAP
+
+        self.execution_dir = os.getcwd()
+        os.chdir("/tmp")
+        retval = self.scan_system()
+        if int(retval) != 0:
```
Comment (pirat89) [inline preupg/application.py]: the `int()` func is pointless here.

File: preupg/application.py [hunk 6]
```diff
@@ -440,11 +442,10 @@ def get_cmd_convertor(self):
...
 
-    def get_proper_scenario(self, scenario):
-        if not self.conf.contents:
-            return scenario
-        scenario = scenario.replace('-results', '')
-        return scenario
+    def rename_custom_module_set(self, module_set_dirname):
+        if self.conf.contents:
+            module_set_dirname = module_set_dirname.replace('-results', '')
+        return module_set_dirname
```
Comment (pirat89) [inline preupg/application.py]: See issue #331. Will be resolved as part of different PR probably.

File: preupg/application.py (hunk 2 above)
Comment (bocekm) [reply preupg/application.py]: Will be dealt with under https://github.com/upgrades-migrations/preupgrade-assistant/issues/332.

File: preupg/application.py (hunk 1 above)
Comment (bocekm) [reply preupg/application.py]: Moved to `determine_module_set_copy_location()`.

File: preupg/application.py (hunk 3 above)
Comment (bocekm) [reply preupg/application.py]: Moved to `determine_module_set_copy_location()`.

File: preupg/application.py (hunk 4 above)
Comment (bocekm) [reply preupg/application.py]: Moved to `preupg/cli.py/resolve_option_dependencies()`.

File: preupg/application.py (hunk 5 above)
Comment (bocekm) [reply preupg/application.py]: Removed.

**Comment by AloisMahdal** [comment]
`qa_ack`: I will write downstream test specifically for `--list-rules`.

**Comment by AloisMahdal** [comment]
Downstream test finished and ran with `preupgrade-assistant-2.5.0-0.201710231614Z.PR330.el6_8`; all went fine except for minor issues, which I have reported separately as #333 and #334.

**Comment by pirat89** [comment]
IMO both issues should be resolved by this PR.

**Comment by bocekm** [comment]
Added fix for https://github.com/upgrades-migrations/preupgrade-assistant/issues/334.

**Comment by bocekm** [comment]
@pirat89, I will certainly not be fixing the #333 issue in this PR. See the rationale here: https://github.com/upgrades-migrations/preupgrade-assistant/issues/333#issuecomment-339693586.

**Comment by AloisMahdal** [comment]
Re-tested with `preupgrade-assistant-2.5.0-0.201710261532Z.PR330.el6_8`.  The exit code is now 22.

File: man/preupg.h2m [hunk 7]
```diff
@@ -1,26 +1,20 @@
...
-preupg [--list-rules] [-s, --scan MODULE_SET]
-
-preupg [--riskcheck] [-v, --verbose]
-
-preupg [-u, --upload URL] [-r, --results TARBALL]
-
-preupg [--cleanup]
+preupg -u|--upload URL [-r|--results TARBALL]
 
 [DESCRIPTION]
```
Comment (jhornice) [inline man/preupg.h2m]: Shouldn't there be a description from https://github.com/upgrades-migrations/preupgrade-assistant/pull/337/files?

Comment (bocekm) [reply man/preupg.h2m]: Yes, thank you. I have created an issue for that: https://github.com/upgrades-migrations/preupgrade-assistant/issues/338.
